src/ts/component/menu/dataview/calendar.tsx

```diff
@@ -144,10 +156,76 @@ const MenuCalendar = observer(class MenuCalendar extends React.Component<I.Menu,
 		const { value } = data;
 
 		this.originalValue = value;
+
+		const selectedCalendarDate = U.Date.getCalendarDateParam(value);
+		this.setState({
+			selectedCalendarDate,
+		});
+
 		this.initDotMap();
+		this.rebind();
 		this.forceUpdate();
 	};
 
+	componentWillUnmount () {
+		this.unbind();
+	};
+
+	rebind () {
+		this.unbind();
+		$(window).on('keydown.menu', e => this.onKeyDown(e));
+	};
+
+	unbind () {
+		$(window).off('keydown.menu');
+	};
+
+	onKeyDown = (e: any) => {
+		e.stopPropagation();
+		keyboard.disableMouse(true);
+
+		keyboard.shortcut('arrowup, arrowdown, arrowleft, arrowright', e, (pressed: string) => {
+			e.preventDefault();
+
+			this.onArrow(pressed as ArrowDirection);
+		});
+
+		const { selectedCalendarDate: selectedDay } = this.state;
+		if (!selectedDay) {
+			return;
+		};
+
+		keyboard.shortcut('enter', e, () => {
+			e.preventDefault();
+
+			this.setOrOpenDate(selectedDay);
+		});
+	};
+
+	onArrow = (dir: ArrowDirection) => {
+		const num = [ ArrowDirection.Up, ArrowDirection.Down ].includes(dir) ? 7 : 1;
+		const d = [ ArrowDirection.Up, ArrowDirection.Left ].includes(dir) ? -1 : 1;
+		const daysDelta = num * d;
+
+		const { param } = this.props;
+		const { data } = param;
+		const { value } = data;
+		const currentMonth = U.Date.getCalendarDateParam(value).m;
+
+		const { selectedCalendarDate } = this.state;
+		const newDateValue = U.Date.timestamp(selectedCalendarDate.y, selectedCalendarDate.m, selectedCalendarDate.d) + daysDelta * 86400 + 86400 / 2;
```

Comment:
`+ 86400 / 2`
Is an easy fix for Daylight Saving Time change. Otherwise it breaks stepping by one day.